proposals/0035-linalg-matrix.md

```diff
@@ -1312,6 +1337,45 @@ two input vectors. The matrix scope can be `Thread`, `Wave`, or `ThreadGroup`.
 The element type of the output matrix matches the element type of the input
 vectors.
 
+#### Pipeline State Validation Metadata
+
+Shader Model 6.10 will introduce a version 4 of the Pipeline
+State Validation RuntimeInfo structure. A new 32-bit unsigned integer
+`LinalgMatrixUses` will count the number of `MatrixUse` objects appended after
+the signature output vectors (presently the last data at the end of `PSV0` for
+version 3).
+
+The `MatrixUse` object is defined:
+
+```c
+struct MatrixUse {
+  uint32_t Dimensions[3]; // M, N, K
+  uint32_t Scope;
+  uint32_t Flags; // do we need this?
+  uint32_t OperandType;
+  uint32_t ResultType;

```

Comment:
I'd be tempted to bit-pack a bit more here. Maybe premature, but dimensions can't exceed `uint16_t`, and similarly the other types are enums with < 255 entries so could be `uint8_t`.

Reply:
I'm not sure we've settled on a maximum bound for the M or N dimensions. `uint16_t` is probably reasonable for thread and wave scope, but there are definitely larger matrices we may want to support with thread group scope that could be well in excess of that. I'm hesitant to put us into any corners.

The scope and type enums could definitely likely be smaller safely. I'll take a stab at fleshing out the enum values for the type enumeration and condensing the structure a bit. We do need it to be an even number of dwords so that we don't mess up alignment when packing.

Reply:
Oh right, there's only maximums for K, so maybe we keep those as `uint32_t[3]` then, and then just pack the enums together. That should all fit in one dword, making the total structure 4 dwords / 16 bytes.